Ginger/GingerCore/Drivers/SeleniumDriver.cs

```diff
@@ -5301,57 +5318,148 @@ private void DoGetRecordings()
+
+                                if (LearnAdditionalDetails)
+                                {
+                                    string xCordinate = PLR.GetValueString();
+                                    string yCordinate = PLR.GetValueString();
+                                    ElementInfo eInfo = LearnRecorededElementFullDetails(xCordinate, yCordinate);
+
+                                    if (eInfo != null)
+                                    {
+                                        args.LearnedElementInfo = eInfo;
+                                    }
+                                }
+                                if (ElementRecorded != null)
+                                {
+                                    //New implementation supporting POM
+                                    OnLearnedElement(args);
+                                }
+                                else
+                                {  
+                                    //Temp existing implementation
+                                    ActUIElement actUI = new ActUIElement();
+                                    actUI.Description = GetDescription(args.Operation, args.LocateValue, args.ElementValue, Convert.ToString(args.Type));
+                                    actUI.ElementLocateBy = GetLocateBy(Convert.ToString(args.LocateBy));
+                                    actUI.ElementLocateValue = args.LocateValue;
+                                    actUI.ElementType = GetElementTypeEnum(null, Convert.ToString(args.Type)).Item2;
+                                    if (Enum.IsDefined(typeof(ActUIElement.eElementAction), args.Operation))
+                                        actUI.ElementAction = (ActUIElement.eElementAction)Enum.Parse(typeof(ActUIElement.eElementAction), args.Operation);
+                                    else
+                                        continue;
+                                    actUI.Value = args.ElementValue;
+                                    this.BusinessFlow.AddAct(actUI);
+                                    if (mActionRecorded != null)
+                                    {
+                                        mActionRecorded.Invoke(this, new POMEventArgs(Driver.Title, actUI));
+                                    }
+                                }                                
                             }
                         }
                     }
+                    catch (Exception e)
+                    {
+                        Reporter.ToLog(eLogLevel.ERROR, "Error occured while recording", e);
+                    }
                 }
             }
             catch (Exception e)
             {
-                Reporter.ToLog(eLogLevel.ERROR, e.Message);
+                Reporter.ToLog(eLogLevel.ERROR, "Error occured while recording", e);
             }
         }
 
+        public bool LearnAdditionalDetails { get; set; }
```

Comment:
Let's try to minimize the amount of fields/events we adding to each driver:
1- please remove LearnAdditionalDetails as global field and add it to StartRecording(bool LearnAdditionalDetails) so it will passed to Driver as part of start recording call params

2- please merge the 2 events into one event called "RecordingEvent" and in it eventArgs add "EventType" enum which  will have "ElementRecorded, PageChanged, <<whatEverWeWillNeedLater>>", and "EventObj" field which will be object field to pass any information will be needed as part of the event.

Reply:
Both the points are good to have but now it will require lot many changes,
#1 - for this we will have to add this parameter in the DriverBase class and simultaneously in all drivers where this method is overrided, this is we require since we are supporting the existing implementation currently.

#2 - we shouldnt combine the events as both are having there own separate functionality, objects will be reduced but it would not be in good readable way if we combine.

Please let me know if you think we still need to implement them, i will proceed.